[Win32] gmake removes perly.c when building 5.39.5

This one feels like a pebcak, but I haven't so far been able to nail it.
No such problem with 5.39.4 and earlier.
For simplicity, I've created an entirely new folder (C:\New) into which I unpack perl-5.39.5.tar.gz.
Then 'cd' to C:\New\perl-5.39.5\win32 and run <code>gmake CCHOME=C:\winlibs-ucrt-1320\mingw64</code>
Perl builds fine, using the default GNUmakefile values, but the process terminates with:
```
...
gmake[1]: Leaving directory 'C:/New/perl-5.39.5/cpan/podlators'
Running pm_to_blib for cpan/version directly
rm ..\perly.c
```
Any ideas as to what the cause might be ?
If I don't provide the "CCHOME" argument, the build throws an error ... but still manages to <code>rm ..\perly.c</code> before terminating.

If I want to then run <code>gmake test</code> I first have to reinstall perly.c - which again needs to be reinstalled before running <code>gmake install</code>.

As a check, I've also built a freshly unpacked 5.39.4 source using the same toolset and the same gmake command. That works fine and perly.c does not get clobbered. The process terminates correctly with:
```
gmake[1]: Leaving directory 'C:/New/perl-5.39.4/cpan/podlators'
Running pm_to_blib for cpan/version directly
```
I'll have another crack at it tomorrow. In the meantime, any ideas are welcome.

Here's the <code>perl -V</code> output  for the perl-5.39.5 build:
```
C:\New\perl-5.39.5\win32>..\perl -I..\lib -V
Summary of my perl5 (revision 5 version 39 subversion 5) configuration:

  Platform:
    osname=MSWin32
    osvers=10.0.22621.2715
    archname=MSWin32-x64-multi-thread
    uname=''
    config_args='undef'
    hint=recommended
    useposix=true
    d_sigaction=undef
    useithreads=define
    usemultiplicity=define
    use64bitint=define
    use64bitall=undef
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
  Compiler:
    cc='gcc'
    ccflags =' -DWIN32 -DWIN64 -DPERL_TEXTMODE_SCRIPTS -DMULTIPLICITY -DPERL_IMPLICIT_SYS -DUSE_PERLIO -D__USE_MINGW_ANSI_STDIO -fwrapv -fno-strict-aliasing -mms-bitfields'
    optimize='-Os'
    cppflags='-DWIN32'
    ccversion=''
    gccversion='13.2.0'
    gccosandvers=''
    intsize=4
    longsize=4
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='long long'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='g++'
    ldflags ='-s -L"c:\perl\lib\CORE" -L"C:\winlibs-ucrt-1320\mingw64\lib" -L"C:\winlibs-ucrt-1320\mingw64\x86_64-w64-mingw32\lib" -L"C:\winlibs-ucrt-1320\mingw64\lib\gcc\x86_64-w64-mingw32\13.2.0"'
    libpth=C:\winlibs-ucrt-1320\mingw64\lib C:\winlibs-ucrt-1320\mingw64\x86_64-w64-mingw32\lib C:\winlibs-ucrt-1320\mingw64\lib\gcc\x86_64-w64-mingw32\13.2.0
    libs= -lmoldname -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladvapi32 -lshell32 -lole32 -loleaut32 -lnetapi32 -luuid -lws2_32 -lmpr -lwinmm -lversion -lodbc32 -lodbccp32 -lcomctl32
    perllibs= -lmoldname -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladvapi32 -lshell32 -lole32 -loleaut32 -lnetapi32 -luuid -lws2_32 -lmpr -lwinmm -lversion -lodbc32 -lodbccp32 -lcomctl32
    libc=
    so=dll
    useshrplib=true
    libperl=libperl539.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_win32.xs
    dlext=dll
    d_dlsymun=undef
    ccdlflags=' '
    cccdlflags=' '
    lddlflags='-shared -s -L"c:\perl\lib\CORE" -L"C:\winlibs-ucrt-1320\mingw64\lib" -L"C:\winlibs-ucrt-1320\mingw64\x86_64-w64-mingw32\lib" -L"C:\winlibs-ucrt-1320\mingw64\lib\gcc\x86_64-w64-mingw32\13.2.0"'


Characteristics of this binary (from libperl):
  Compile-time options:
    HAS_LONG_DOUBLE
    HAS_TIMES
    HAVE_INTERP_INTERN
    MULTIPLICITY
    PERLIO_LAYERS
    PERL_COPY_ON_WRITE
    PERL_DONT_CREATE_GVSV
    PERL_HASH_FUNC_SIPHASH13
    PERL_HASH_USE_SBOX32
    PERL_IMPLICIT_SYS
    PERL_MALLOC_WRAP
    PERL_OP_PARENT
    PERL_PRESERVE_IVUV
    PERL_USE_SAFE_PUTENV
    USE_64_BIT_INT
    USE_ITHREADS
    USE_LARGE_FILES
    USE_LOCALE
    USE_LOCALE_COLLATE
    USE_LOCALE_CTYPE
    USE_LOCALE_NUMERIC
    USE_LOCALE_TIME
    USE_PERLIO
    USE_PERL_ATOF
    USE_THREAD_SAFE_LOCALE
  Built under MSWin32
  Compiled at Nov 21 2023 00:18:53
  @INC:
    ..\lib
    C:/New/perl-5.39.5/lib
```

No mention of "yacc" or "bison" anywhere in the build, and I see only the same 2 references to "perly.c" during the build.
The problem with perly.c seems to be avoided if I use make-4.2.1 (which I was using until a week or so ago) instead of make-4.4.1.

But now there's another annoying issue in that <code>make distclean</code> is not cleaning things up like it should, and I have to start from a freshly unpacked source for every mingw-w64 build I do.
When I try to re-use a distcleaned source, the "make" step bails out at:
```
make[1]: Leaving directory 'D:/comp-1320-ucrt/perl-5.39.5/ext/Sys-Hostname'
Makefile.PL: The "xdefine" exists, skipping the configure step.
Use "D:\comp-1320-ucrt\perl-5.39.5\miniperl.exe Makefile.PL --configure"
or: "D:\comp-1320-ucrt\perl-5.39.5\miniperl.exe Makefile.PL --force
to force the configure step.
Warning: No Makefile!
make[1]: Entering directory 'D:/comp-1320-ucrt/perl-5.39.5/dist/Time-HiRes'
make[1]: *** No rule to make target 'all'.  Stop.
make[1]: Leaving directory 'D:/comp-1320-ucrt/perl-5.39.5/dist/Time-HiRes'
make[1]: Entering directory 'D:/comp-1320-ucrt/perl-5.39.5/dist/Time-HiRes'
make[1]: *** No rule to make target 'all'.  Stop.
make[1]: Leaving directory 'D:/comp-1320-ucrt/perl-5.39.5/dist/Time-HiRes'
Unsuccessful make(dist/Time-HiRes): code=512 at ..\make_ext.pl line 584.
make: *** [GNUmakefile:1560: Extensions] Error 2
```
There could still be a pebcak element at play here.
It might take me a while to sort through it all.

--- a/win32/GNUmakefile
+++ b/win32/GNUmakefile
@@ -1904,3 +1904,6 @@ nok: utils $(PERLEXE) $(PERLDLL) Extensions_nonxs Extensions
 
 nokfile: utils $(PERLEXE) $(PERLDLL) Extensions_nonxs Extensions
 	$(PERLEXE) ..\utils\perlbug -nok -s "(UNINSTALLED)" -F perl.nok
+
+# prevent implicit rule
+%.c : %.y

